src/marqo/tensor_search/add_docs.py

```diff
@@ -77,13 +88,16 @@ def threaded_download_images(allocated_docs: List[dict], image_repo: dict,
                                 continue
 
 
-def download_images(docs: List[dict], thread_count: int, non_tensor_fields: Tuple, image_download_headers: dict) -> dict:
+def download_images(docs: List[dict], thread_count: int, tensor_fields: Optional[Tuple],
+                    non_tensor_fields: Optional[Tuple], image_download_headers: dict) -> dict:
     """Concurrently downloads images from each doc, storing them into the image_repo dict
     Args:
         docs: docs with images to be downloaded. These will be allocated to each thread
         thread_count: number of threads to spin up
+        tensor_fields: A tuple of tensor_fields. Images will be downloaded for these fields only. Cannot be provided
+            at the same time as `non_tensor_fields`.
         non_tensor_fields: A tuple of non_tensor_fields. No images will be downloaded for
-            these fields
+            these fields. Cannot be provided at the same time as `tensor_fields`.
         image_download_headers: A dict of image download headers for authentication.
     This should be called only if treat URLs as images is True
```

Comment:
ditto regarding communication about `non_tensor_fields` and `tensor_fields` validation